clients/nexus-client/src/lib.rs

```diff
@@ -382,3 +382,30 @@ impl From<omicron_common::api::internal::shared::ExternalPortDiscovery>
         }
     }
 }
+
+impl From<types::ProducerKind>
+    for omicron_common::api::internal::nexus::ProducerKind
+{
+    fn from(kind: types::ProducerKind) -> Self {
+        use omicron_common::api::internal::nexus::ProducerKind;
+        match kind {
+            types::ProducerKind::SledAgent => ProducerKind::SledAgent,
+            types::ProducerKind::Instance => ProducerKind::Instance,
+            types::ProducerKind::Service => ProducerKind::Service,
+        }
+    }
+}
+
+impl From<types::ProducerEndpoint>
+    for omicron_common::api::internal::nexus::ProducerEndpoint
+{
+    fn from(ep: types::ProducerEndpoint) -> Self {
+        Self {
+            id: ep.id,
+            kind: ep.kind.into(),
+            address: ep.address.parse().unwrap(),
```

Comment:
Changing it to a try from is _not_ annoying, because there's only one place it's used: this PR! I will change it. I thought it was safe because you only get this field from an already progenitor-generated type. But I think that relies a bit more than needed on progenitor internally validating the IP address format. Will update!

Reply:
Done in f44da6fc0